pkg/target/target.go

```diff
@@ -27,20 +27,20 @@ import (
 //   - These are defined by this pattern: [a-z0-9]([-a-z0-9]*[a-z0-9])?
 //   - You'll see that this is the first two-thirds or so of the pattern below
 //
-//   PREFIX-BASED WILDCARDS
+//   PREFIX OR SUFFIX BASED WILDCARDS
 //   - A typical namespace must end in an alphanumeric character.  A prefixed wildcard
-//     can end in "*" (like `kube*`) or "-*" (like `kube-*`).
-//   - To implement this, we add the following: (\*|-\*)?
-//   - Crucially, this _does not_ allow the value to end in a dash (like `kube-`).  That is
-//     not a valid namespace and not a wildcard, so it's disallowed
+//     can end in "*" (like `kube*`) or "-*" (like `kube-*`), and a suffixed wildcard
+//     can start with "*" (like `*system`) or "*-" (like `*-system`).
+//   - To implement this, we add the following (\*|\*-)? as a prefix and (\*|-\*)? as a suffix.
+//   - Crucially, this _does not_ allow the value to start or end in a dash (like `-system` or `kube-`).
+//     That is not a valid namespace and not a wildcard, so it's disallowed
```

Comment:
Does this cover cases like `*-*`? If not, can we call that out too?

Reply:
Thanks for pointing this out 🙂

This change doesn't allow more than one asterisk in the pattern, therefore `*-*` is not allowed. I'll add this case to both regexr and comment.